Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -239,9 +239,24 @@ auto compilationMessageData(convertMessages(
         m_library.label = label;
 }
 
-id<MTLFunction> ShaderModule::getNamedFunction(const String& name) const
+id<MTLFunction> ShaderModule::getNamedFunction(const String& name, const HashMap<String, float>& keyValueReplacements) const
 {
-    return [m_library newFunctionWithName:name];
+    if (!keyValueReplacements.size())
+        return [m_library newFunctionWithName:name];
+
+    MTLFunctionConstantValues *constantValues = [MTLFunctionConstantValues new];
+    for (auto& kvp : keyValueReplacements) {
+        // FIXME: specify actual types from the compiler information
+        [constantValues setConstantValue:&kvp.value type:MTLDataTypeFloat withName:kvp.key];
```

Comment:
I think we should fully implement this, or add FIXMEs with bug links. There are 2 pieces missing:

1. As the comment suggests, MTLDataTypeFloat is not the only type. Ideally, this would be done by the WGSL compiler emitting metadata telling you about, for each entry point, which pipeline-overridable constants exist, and what their types are. I think we should make sure that the WGSL compiler's API signatures support returning this data, even if compiler right now can't actually populate that data structure and it has to be shimmed some other way (possibly MTLReflection). We should hook this data structure up here, rather than using a `HashMap<String, float>` that is not expressive enough.
2. Pipeline-overridable constants in WGSL can be identified in 2 ways: either by name or by index. I think we should hook that up while we're here.

Reply:
I think the second point requires information from the compiler, to tell us the id which corresponds to a given name in the generated metal source as metal has no concept of an id and the ids are not linearly ordered. From the specification, it appears the IDs are basically secondary names:
```
@id(0)      override has_point_light: bool = true;  // Algorithmic control.
@id(1200)   override specular_param: f32 = 2.3;     // Numeric control.
@id(1300)   override gain: f32;                     // Must be overridden.
```

Reply:
I filed https://bugs.webkit.org/show_bug.cgi?id=250444 for the first issue and will add a FIXME, but we can also do this in the mean time by calling -[MTLLibrary newFunctionWithName:]` twice and using `-[MTLFunction functionConstantsDictionary]`